src/core/net/ip6.cpp

```diff
@@ -66,6 +68,7 @@ void Ip6::Init(void)
 {
     sMpl = new(&sMplBuf) Mpl;
     sForwardingEnabled = false;
+    sIsReceiveIp6FilterEnabled = false;
```

Comment:
Personally, I'd prefer filtering enabled by default, since only in debugging cases would it be off, but I don't think it matters much.